container-core/src/test/java/com/yahoo/jdisc/http/server/jetty/ConnectorFactoryTest.java

```diff
@@ -16,46 +18,73 @@
 import java.util.Map;
 
 import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertEquals;
 
 /**
  * @author Einar M R Rosenvinge
+ * @author bjorncs
  */
 public class ConnectorFactoryTest {
 
-    @Test
-    public void requireThatServerCanBindChannel() throws Exception {
-        Server server = new Server();
+    private Server server;
+
+    @Before
+    public void createServer() {
+        server = new Server();
+    }
+
+    @After
+    public void stopServer() {
         try {
-            ConnectorConfig config = new ConnectorConfig(new ConnectorConfig.Builder());
-            ConnectorFactory factory = createConnectorFactory(config);
-            JettyConnectionLogger connectionLogger = new JettyConnectionLogger(
-                    new ServerConfig.ConnectionLog.Builder().enabled(false).build(),
-                    new VoidConnectionLog());
-            DummyMetric metric = new DummyMetric();
-            var connectionMetricAggregator = new ConnectionMetricAggregator(new ServerConfig(new ServerConfig.Builder()), metric);
-            JDiscServerConnector connector =
-                    (JDiscServerConnector)factory.createConnector(metric, server, connectionLogger, connectionMetricAggregator);
-            server.addConnector(connector);
-            server.setHandler(new HelloWorldHandler());
-            server.start();
-
-            SimpleHttpClient client = new SimpleHttpClient(null, connector.getLocalPort(), false);
-            SimpleHttpClient.RequestExecutor ex = client.newGet("/blaasdfnb");
-            SimpleHttpClient.ResponseValidator val = ex.execute();
-            val.expectContent(equalTo("Hello world"));
-        } finally {
-            try {
-                server.stop();
-            } catch (Exception e) {
-                //ignore
-            }
+            server.stop();
+            server = null;
+        } catch (Exception e) {
+            //ignore
         }
     }
 
+    @Test
+    public void requireThatServerCanBindChannel() throws Exception {
+        ConnectorConfig config = new ConnectorConfig(new ConnectorConfig.Builder());
+        ConnectorFactory factory = createConnectorFactory(config);
+        JDiscServerConnector connector = createConnectorFromFactory(factory);
+        server.addConnector(connector);
+        server.setHandler(new HelloWorldHandler());
+        server.start();
+
+        SimpleHttpClient client = new SimpleHttpClient(null, connector.getLocalPort(), false);
+        SimpleHttpClient.RequestExecutor ex = client.newGet("/blaasdfnb");
+        SimpleHttpClient.ResponseValidator val = ex.execute();
+        val.expectContent(equalTo("Hello world"));
+    }
+
+    @Test
+    public void constructed_connector_is_based_jdisc_connector_config() {
```

Comment:
Suspect you missed _on .....based_on_jdisk...

Reply:
Fixed